src/ui/public/agg_types/metrics/top.js

```diff
@@ -0,0 +1,59 @@
+import { noop, get } from 'lodash';
+import MetricAggTypeProvider from 'ui/agg_types/metrics/metric_agg_type';
+import topEditor from 'ui/agg_types/controls/top.html';
+
+export default function AggTypeMetricTopProvider(Private) {
+  const MetricAggType = Private(MetricAggTypeProvider);
+
+  return new MetricAggType({
+    name: 'top_hits',
+    title: 'Top',
+    makeLabel: function (aggConfig) {
+      const prefix = aggConfig.params.sortOrder.val === 'desc' ? 'Latest' : 'Earliest';
```

Comment:
Ah forgot there was that custom label already ;o)
What about First/Last ? Although I am fine with your proposition.